File: app/src/main/java/com/example/testproject/network/CallBack.java
package com.example.testproject.network;


import com.example.testproject.BuildConfig;
import com.example.testproject.util.LogUtil;
import com.example.testproject.util.ToastUtil;

/**
 * Created by mike.li.
 */

public abstract class CallBack<T> {
    public abstract void onResponse(T response);
    //返回进度
    public void onProgress(long readLength, long contentLength){
        LogUtil.i("percent:"+readLength+":"+contentLength);
    }
    public void onError(final Throwable throwable){
        if(BuildConfig.DEBUG){
            throwable.printStackTrace();
        }
        ToastUtil.showToast(throwable.getMessage());
    }
}

File: app/src/main/java/com/example/testproject/network/CallBack.kt
package com.example.testproject.network


import com.example.testproject.BuildConfig
import com.example.testproject.util.LogUtil
import com.example.testproject.util.ToastUtil

/**
 * Created by mike.li.
 */

abstract class CallBack<T> {
    abstract fun onResponse(response: T)
    //返回进度
    fun onProgress(readLength: Long, contentLength: Long) {
        LogUtil.i("percent:$readLength:$contentLength")
    }

    open fun onError(throwable: Throwable) {
        if (BuildConfig.DEBUG) {
            throwable.printStackTrace()
        }
        throwable.message?.let { ToastUtil.showToast(it) }
    }
}
